### ISSUE TITLE
Add support for Implicit Results in OracleQueryRunner

### ISSUE THREAD

**Issue by adetante** [typeorm/typeorm#8046]
### The Problem

Oracle Implicit Results allow queries in PL/SQL to be returned to Node.js without requiring REF CURSORS or bind variables.
Implicit Results are supported by Node-oracledb library (please see https://oracle.github.io/node-oracledb/doc/api.html#implicitresults), but not supported in TypeORM OracleQueryRunner.

For example, with this schema:

```sql
CREATE TABLE EXAMPLES(
    id NUMBER GENERATED BY DEFAULT AS IDENTITY,
    name VARCHAR2(50) NOT NULL,
    PRIMARY KEY(id)
);

INSERT INTO EXAMPLES (name) VALUES ('Test 1');
INSERT INTO EXAMPLES (name) VALUES ('Test 2');
INSERT INTO EXAMPLES (name) VALUES ('Test 3');
INSERT INTO EXAMPLES (name) VALUES ('Test 4');

CREATE OR REPLACE PROCEDURE GET_EXAMPLES(min_id NUMBER)
AS
    cur SYS_REFCURSOR;
BEGIN
    open cur for
    SELECT * FROM EXAMPLES WHERE id >= min_id;
    DBMS_SQL.return_result(cur);
END;
```

I can use the following NodeJS code to call the Stored Procedure and extract the result

```js
const oracledb = require('oracledb');

const connection = await oracledb.getConnection({
    user          : "***",
    password      : "***",
    connectString : "***"
  });

const result = await connection.execute(
  `begin
      GET_EXAMPLES(:min_id);
  end;`,
  {
    min_id: 2,
  }
);
const { implicitResults } = result; // implicitResults contains the data
```

But this `implicitResults` value is not returned by the TypeORM API:

```ts
import { getConnection } from 'typeorm';
import * as oracledb from 'oracledb';

// ...
const connection = getConnection();
const result = await connection.query(
`
    begin
      GET_EXAMPLES_2(:min_id);
    end;
`,
[
  { dir: oracledb.BIND_IN, val: 3 }
]);
// result is undefined
```

### The Solution

Add support for `implicitResults` in OracleQueryRunner.

### Considered Alternatives

* Use directly the `oracledb` library
* Use OUT parameter instead of implicit results in Stored Procedures

### Relevant Database Driver(s)

| DB Type              | Relevant      |
| ---                  | ---           |
| `aurora-data-api`    | no            |
| `aurora-data-api-pg` | no            |
| `better-sqlite3`     | no            |
| `cockroachdb`        | no            |
| `cordova`            | no            |
| `expo`               | no            |
| `mongodb`            | no            |
| `mysql`              | no            |
| `nativescript`       | no            |
| `oracle`             | yes            |
| `postgres`           | no            |
| `react-native`       | no            |
| `sap`                | no            |
| `sqlite`             | no            |
| `sqlite-abstract`    | no            |
| `sqljs`              | no            |
| `sqlserver`          | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✅ Yes, I have the time, and I know how to start.

**Comment by imnotjames** [comment]
Hey, that's really cool!

**Comment by pleerock** [comment]
@adetante thanks for the contribution! Closed via #8050
